tools/generate_plugindata/main.go

```diff
@@ -0,0 +1,117 @@
+// generate_plugindata is a tool used to inject the sample configuration into all the plugins
+// It extracts the sample configuration from the plugins README.md
+// Then the plugin's main source file is used as a template, and {{ .SampleConfig }} is replaced with the configuration
+// This tool is then also used to revert these changes with the `--clean` flag
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log" //nolint:revive
+	"os"
+	"text/template"
+
+	"github.com/gomarkdown/markdown"
+	"github.com/gomarkdown/markdown/ast"
+	"github.com/gomarkdown/markdown/parser"
+)
+
+// extractPluginData reads the README.md to get the sample configuration
+func extractPluginData() (string, error) {
+	readMe, err := os.ReadFile("README.md")
+	if err != nil {
+		return "", err
+	}
+	extensions := parser.CommonExtensions | parser.AutoHeadingIDs
+	mdParser := parser.NewWithExtensions(extensions)
+	md := markdown.Parse(readMe, mdParser)
+
+	var currentSection string
+	for _, t := range md.GetChildren() {
+		switch tok := t.(type) {
+		case *ast.Heading:
+			currentSection = tok.HeadingID
+		case *ast.CodeBlock:
+			if currentSection == "configuration" && string(tok.Info) == "toml" {
+				return string(tok.Literal), nil
+			}
+		}
+	}
+
+	fmt.Printf("No configuration found for plugin: %s\n", os.Getenv("GOPACKAGE"))
+
+	return "", nil
```

Comment:
if we get this far should we be returning an error that no configuration was found?

Reply:
I originally had it fail, and the issue I ran into was that you would end up with a bunch of partially generated files up to the plugin it failed to find the configuration of. It seemed better if it just warned about it, moved on, so that the cleaning step would run automatically.

Reply:
I originally had it fail, and the issue I ran into was that you would end up with a bunch of partially generated files up to the plugin it failed to find the configuration of. It seemed better if it just warned about it, moved on, so that the cleaning step would run automatically.

Reply:
I'm ok having the linter make sure it's there, then letting this code grab it without much error checking.